pbj-core/pbj-runtime/src/main/java/com/hedera/pbj/runtime/io/buffer/BufferedData.java

```diff
@@ -31,22 +32,47 @@ public class BufferedData implements BufferedSequentialData, ReadableSequentialD
     /** {@link ByteBuffer} used as backing buffer for this instance */
     private final ByteBuffer buffer;
 
+    private final boolean direct;
+
     /**
      * Wrap an existing allocated {@link ByteBuffer}. No copy is made.
      *
      * @param buffer the {@link ByteBuffer} to wrap
+     * @param direct If this is a dirrect buffer allocation.
      */
-    private BufferedData(@NonNull final ByteBuffer buffer) {
+    private BufferedData(@NonNull final ByteBuffer buffer, final boolean direct) {
         this.buffer = buffer;
+        this.direct = direct;
         // I switch the buffer to BIG_ENDIAN so that all our normal "get/read" methods can assume they are in
         // BIG_ENDIAN mode, reducing the boilerplate around those methods. This necessarily means the LITTLE_ENDIAN
         // methods will be slower. I'm assuming BIG_ENDIAN is what we want to optimize for.
         this.buffer.order(BIG_ENDIAN);
     }
 
+    /**
+     * Wrap an existing allocated {@link ByteBuffer}. No copy is made.
+     *
+     * @param buffer the {@link ByteBuffer} to wrap
+     */
+    private BufferedData(@NonNull final ByteBuffer buffer) {
+        this(buffer, false);
+    }
+
     // ================================================================================================================
     // Static Builder Methods
 
+    /**
+     * Wrap an existing allocated {@link ByteBuffer}. No copy is made.
+     *
+     * @param buffer the {@link ByteBuffer} to wrap
+     * @param direct If this is a direct buffer allocation.
+     * @return new instance using {@code buffer} as its data buffer
+     */
+    @NonNull
+    public static BufferedData wrap(@NonNull final ByteBuffer buffer, final boolean direct) {
```

Comment:
We do not need to do this, can just ask the byte buffer if it is direct

Reply:
OK. I'll try to address this and have a flag. I wa just trying to avoid the perf hit when asking (especially in the case of parsing).